extensions/amp-mega-menu/0.1/test/test-amp-mega-menu.js

```diff
@@ -82,7 +82,42 @@ describes.realWin(
       const headingClass = '.i-amphtml-mega-menu-heading';
       expect(element.querySelectorAll(headingClass).length).to.equal(3);
       const contentClass = '.i-amphtml-mega-menu-content';
-      expect(element.querySelectorAll(contentClass).length).to.equal(3);
+      expect(element.querySelectorAll(contentClass).length).to.equal(2);
+    });
+
+    it('should add event listeners on component layout', async () => {
```

Comment:
You should test behaviors rather than the structure of the code.

For example, you should have a test that checks when you click an item that the menu opens rather than the fact that a function was called.

Split this up into separate tests testing the behavior instead. You should also check the publicly observable outcome of an action rather than the internal code organization. For example, instead of checking that `collapse` is called which you hit escape, check that the menu is open via some CSS selector, then check that it is closed via some CSS selector.